engine/grammar/allkeys_test.go

```diff
@@ -14,8 +14,9 @@ func TestAllKeysMatcher(t *testing.T) {
 	dto := &dtos.MatcherDTO{
 		MatcherType: "ALL_KEYS",
 	}
+	ruleBuilder := NewRuleBuilder(nil, nil, nil, SyncProxyFeatureFlagsRules, SyncProxyRuleBasedSegmentRules, logger)
```

Comment:
what about matcher receive the rule builder so then you only call the BuildMatcher>